### PR TITLE
Add kubebuilder boilerplate

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Set up project structure and add kubebuilder boilerplate.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# More info: https://docs.docker.com/engine/reference/builder/#dockerignore-file
```
Comment (ahg-g) [inline .dockerignore:1]: can we just reference `.gitignore` instead? see https://github.com/kubernetes-sigs/kueue/blob/main/.dockerignore

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager main.go
```
Comment (ahg-g) [inline Dockerfile:24]: not for later: probably better to have a single place for build, so this should point to the build rule in the makefile, we can do those changes as followups

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (ahg-g) [inline .gitignore:1]: can we just copy the content of https://github.com/kubernetes-sigs/kueue/blob/main/.gitignore

File: hack/boilerplate.go.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/*
```
Comment (ahg-g) [inline hack/boilerplate.go.txt:1]: we need to use the k8s one: https://github.com/kubernetes-sigs/kueue/blob/ee2a9a2634489c7ad449b390d8640aa7a739258e/main.go#L1-L15

Comment (ahg-g) [reply hack/boilerplate.go.txt:1]: ditto all other places

Comment (danielvegamyhre) [reply hack/boilerplate.go.txt:1]: Updated the copyrights

File: .gitignore (hunk 3 above)
Comment (danielvegamyhre) [reply .gitignore]: Done

File: .gitignore [hunk 5]
```diff
@@ -1,26 +1,32 @@
...
 *~
+
+# vendor files
+vendor
+
+# Local Netlify folder
+.netlify
+
+# Directory with CRDs for dependencies
+dep-crds
```
Comment (ahg-g) [inline .gitignore:32]: add line?

File: api/v1alpha1/groupversion_info.go [hunk 6]
```diff
@@ -1,12 +1,9 @@
 /*
-Copyright 2023.
-
+Copyright 2021 The Kubernetes Authors.
```
Comment (ahg-g) [inline api/v1alpha1/groupversion_info.go:2]: 2023

**Comment by ahg-g** [comment]
can you squash into one, the first commit should look nice and tidy :)

File: PROJECT [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Code generated by tool. DO NOT EDIT.
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: batch.x-k8s.io
+layout:
+- go.kubebuilder.io/v3
+projectName: jobset
+repo: batch.x-k8s.io/jobset
```
Comment (ahg-g) [inline PROJECT:9]:
```suggestion
repo: sigs.k8s.io/jobset
```

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,70 @@
+module batch.x-k8s.io/jobset
```
Comment (ahg-g) [inline go.mod:1]:
```suggestion
module sigs.k8s.io/jobset
```

File: api/v1alpha1/groupversion_info.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "jobset.batch.x-k8s.io", Version: "v1alpha1"}
```
Comment (ahg-g) [inline api/v1alpha1/groupversion_info.go:26]:
```suggestion
	GroupVersion = schema.GroupVersion{Group: "batch.x-k8s.io", Version: "v1alpha1"}
```

File: config/crd/bases/jobset.batch.x-k8s.io_jobsets.yaml [hunk 10]
```diff
@@ -0,0 +1,50 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.11.1
+  creationTimestamp: null
+  name: jobsets.jobset.batch.x-k8s.io
+spec:
+  group: jobset.batch.x-k8s.io
```
Comment (ahg-g) [inline config/crd/bases/jobset.batch.x-k8s.io_jobsets.yaml:10]: I think you need to regenerate the artifacts to update this

File: config/rbac/jobset_editor_role.yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    app.kubernetes.io/name: clusterrole
+    app.kubernetes.io/instance: jobset-editor-role
+    app.kubernetes.io/component: rbac
+    app.kubernetes.io/created-by: jobset
+    app.kubernetes.io/part-of: jobset
+    app.kubernetes.io/managed-by: kustomize
+  name: jobset-editor-role
+rules:
+- apiGroups:
+  - jobset.batch.x-k8s.io
```
Comment (ahg-g) [inline config/rbac/jobset_editor_role.yaml:15]: ditto; this needs updating too

File: controllers/jobset_controller.go [hunk 12]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (ahg-g) [inline controllers/jobset_controller.go:1]: we need to move this under pkg, we can do it as follow up.

Comment (danielvegamyhre) [reply controllers/jobset_controller.go]: I went ahead and just moved it really quick

**Pull request merged**